run_southafrica.py

```diff
@@ -1,95 +1,129 @@
-
-if args.webpage == 1:
-    matches_constit = new_za.run_constitution(keywords)
-    specs = 'SouthAfrica-constit'
-    if len(matches_constit) > 0:
-        print(matches_constit) 
-        new_za.print_matches(matches_constit,specs)
-        new_za.delete_no_matches(specs,path=new_za.downloadPath+'/constit')
-    else:
-        new_za.delete_unneeded_files('duplicates-nomatch-'+specs,[],files_path=new_za.downloadPath+'/constit')
-    new_za.delete_unneeded_files('duplicates-'+specs,[])
-    new_za.teardown()
-elif args.webpage == 2: 
-    matches_mandates = new_za.run_mandates(keywords)
-    specs = 'SouthAfrica-mandates'
-    if len(matches_mandates) > 0:
-        print(matches_mandates) 
-        new_za.print_matches(matches_mandates,specs)
-        new_za.delete_no_matches(specs,path=new_za.downloadPath+'/mandates')
+docstring = """
+Extract legislation PDFs from the website for
+South African Parliament.
+"""
+
+webpage_help = """
+Select from:
+1=Constitution
+2=Mandates
+3=Acts
+4=Other Bills
+"""
+
+
+def parse_args(args):
+    """Parse command line parameters
+
+    Args:
+      args ([str]): command line parameters as list of strings
+
+    Returns:
+      :obj:`argparse.Namespace`: command line parameters namespace
+    """
+    parser = argparse.ArgumentParser(description=docstring)
+    parser.add_argument('webpage', metavar='webpage', type=int,
+                        help='Webpage to process.' + webpage_help)
+    parser.add_argument('--driver', '-d',
+                        default=ChromeDriverManager().install(),
+                        help='Path for Chromedriver')
+    parser.add_argument('--path', '-p',
+                        default=None,
+                        help='Path for PDF downloads')
+
+    args = parser.parse_args()
+    return args
+
+
+def main(args):
+    """Main entry point allowing external calls
+
+    Args:
+      args ([str]): command line parameter list
+    """
+    args = parse_args(args)
+    if args.path is not None:
+        download_path = str(args.path)
     else:
-        new_za.delete_unneeded_files('duplicates-nomatch-'+specs,[],files_path=new_za.downloadPath+'/mandates')
-    new_za.delete_unneeded_files('duplicates-'+specs,[])
-    new_za.teardown()
-elif args.webpage == 3: 
-    matches_acts = new_za.run_acts(keywords)
-    specs = 'SouthAfrica-acts'
-    if len(matches_acts) > 0:
-        print(matches_acts) 
-        new_za.print_matches(matches_acts,specs)
-        new_za.delete_no_matches(specs,path=new_za.downloadPath+'/acts')
+        current_dir = Path(os.path.abspath(os.curdir))
+        download_path = current_dir / 'south_africa_output'
+        download_path = str(download_path)  # class wants strings for paths
+        if not os.path.exists(download_path):
+            os.mkdir(download_path)
+
+    # setting up options for Chromedriver
+    # mostly to ensure any PDF links automatically
+    # download the PDFs to download_path
+    options = webdriver.ChromeOptions()
```

Comment:
similarly: I think it's fine to refactor how `options` is handled, but since these options aren't different among run_X.py files, on the `working` branch I wound up creating a way to save this within the `legisWeb` class.